src/Service/ScopeHintService.php

```diff
@@ -24,12 +24,22 @@ public function __construct(
 
     public function getAllScopes() : array
     {
-        //$configArray = $this->treePaths->generate($this->scopeConfig->getValue(null));
+        $configArray = $this->scopeConfig->getValue(null);
 
-        //$all_configs = $this->scopeConfig->getValue(null);
+        $result = array();
+        $this->displayArrayRecursively($configArray, $result, "");
+        return $result;
+    }
 
+    function displayArrayRecursively($array, &$result_array, $path) : void {
```

Comment:
In general, I try to avoid variables as references and use return values instead. Usually (including this case), it's more readable.
Also please add types for the input parameters.